src/main/core/DependencyInjection/Compiler/GeoipPass.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Claroline Connect package.
+ *
+ * (c) Claroline Consortium <consortium@claroline.net>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Claroline\CoreBundle\DependencyInjection\Compiler;
+
+use GeoIp2\Database\Reader;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Removes the optional {@link Reader} service if the geoip database is not available.
+ */
+class GeoipPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (!(new Filesystem())->exists($container->getParameter('claroline.param.geoip_db_path'))) {
+            $container->removeDefinition(Reader::class);
```

Comment:
Why not remove our `MaxMindGeoIpInfoProvider`. It's already optional.

Reply:
right, done